core/src/main/scala/kafka/server/MetadataCache.scala

```diff
@@ -17,91 +17,115 @@
 
 package kafka.server
 
-import kafka.cluster.{EndPoint, BrokerEndPoint, Broker}
+import kafka.cluster.{EndPoint, Broker}
 import kafka.common.TopicAndPartition
 
 import kafka.api._
-import kafka.controller.KafkaController.StateChangeLogger
-import kafka.controller.LeaderIsrAndControllerEpoch
-import org.apache.kafka.common.errors.{ReplicaNotAvailableException, LeaderNotAvailableException}
+import kafka.controller.{KafkaController, LeaderIsrAndControllerEpoch}
+import org.apache.kafka.common.Node
 import org.apache.kafka.common.protocol.{Errors, SecurityProtocol}
-import org.apache.kafka.common.requests.UpdateMetadataRequest
+import org.apache.kafka.common.requests.{MetadataResponse, UpdateMetadataRequest}
 import org.apache.kafka.common.requests.UpdateMetadataRequest.PartitionState
-import scala.collection.{Seq, Set, mutable}
+import scala.collection.{mutable, Seq, Set}
 import scala.collection.JavaConverters._
 import kafka.utils.Logging
 import kafka.utils.CoreUtils._
 
 import java.util.concurrent.locks.ReentrantReadWriteLock
 
-
 /**
  *  A cache for the state (e.g., current leader) of each partition. This cache is updated through
  *  UpdateMetadataRequest from the controller. Every broker maintains the same cache, asynchronously.
  */
 private[server] class MetadataCache(brokerId: Int) extends Logging {
-  private val cache: mutable.Map[String, mutable.Map[Int, PartitionStateInfo]] =
-    new mutable.HashMap[String, mutable.Map[Int, PartitionStateInfo]]()
-  private var aliveBrokers: Map[Int, Broker] = Map()
+  private val stateChangeLogger = KafkaController.stateChangeLogger
+  private val cache = new mutable.HashMap[String, mutable.Map[Int, PartitionStateInfo]]()
```

Comment:
We could replace `new mutable.HashMap` with `mutable.Map` to be consistent with the other map instantiation.